packages/enketo-express/test/client/encryptor.spec.js

```diff
@@ -242,5 +270,84 @@ describe('Encryptor', () => {
                 })
                 .catch(done);
         });
+
+        it('encodes submission XML with non-ASCII characters into UTF-8 before MD5 hashing', (done) => {
+            const record = {
+                xml: '<root>café résumé naïve</root>',
+                instanceId: '1a2b',
+            };
+
+            encryptor
+                .encryptRecord(form, record)
+                .then((encryptedRecord) => {
+                    expect(encryptedRecord.files[0].md5).to.equal(
+                        '2ba225a35204382b2195ac4fcd60a3f9'
+                    );
+                    done();
+                })
+                .catch(done);
+        });
+
+        it('encodes the raw signature with non-ASCII characters into UTF-8 before MD5 hashing', (done) => {
+            form.id = 'form-测试-тест'; // A form ID with non-ASCII characters
+
+            const record = {
+                xml: '<root>this is a record</root>',
+                instanceId: '1a2b',
+            };
+            const mockBytes = forge.util.createBuffer(
+                new Uint8Array(32).fill(65),
+                'raw'
+            );
+            const mockPublicKey = {
+                encrypt: (byteString) => {
+                    const md = forge.md.md5.create();
+                    md.update(byteString);
+                    const hash = md.digest().getBytes();
+                    return String.fromCharCode(
+                        ...hash.split('').map((c) => c.charCodeAt(0) + 66)
+                    );
+                },
+            };
+
+            sandbox.stub(forge.random, 'getBytesSync').returns(mockBytes);
+            sandbox.stub(forge.pki, 'publicKeyFromPem').returns(mockPublicKey);
+
+            const encryptedSymmetricKey = mockPublicKey.encrypt(mockBytes);
+            const expectedElements = [
+                form.id,
+                form.version,
+                forge.util.encode64(encryptedSymmetricKey),
+                record.instanceId,
+                'submission.xml::' +
+                    forge.md.md5
+                        .create()
+                        .update(forge.util.encodeUtf8(record.xml))
+                        .digest()
+                        .toHex(),
+            ];
```

Comment:
Consider using a string template instead of concatenation or our linting rules will complain! 🙇🏻

```suggestion
            const md5 = forge.md.md5
                .create()
                .update(forge.util.encodeUtf8(record.xml))
                .digest()
                .toHex();

            const expectedElements = [
                form.id,
                form.version,
                forge.util.encode64(encryptedSymmetricKey),
                record.instanceId,
                `submission.xml::${md5}`
            ];
```